src/main/java/org/owasp/esapi/reference/DefaultHTTPUtilities.java

```diff
@@ -1097,5 +1139,38 @@ private String decryptString(String ciphertext) throws EncryptionException {
         PlainText plaintext = ESAPI.encryptor().decrypt(restoredCipherText);
         return plaintext.toString();
     }
-}
 
+    /* Helper method, currently used only to check if we allow anonymous
+     * (i.e., unauthenticated) users to perform file uploads.
+     * The default is do allow anonymous users. We start to see if the property
+     * specified by "HttpUtilities.FileUploadAllowAnonymousUser" is set to
+     * {@code true} (which is the default). If that is true, we immediately
+     * return true as no further authentication check is required. However, if
+     * that property is false, then we check if the user is authenticated (i.e.,
+     * the returned {@code User} object is not null).
+     */
+    private boolean allowUserFileUploadAccess() {
+        if ( fileUploadAllowAnonymousUsers ) {
+            return true;    // Okay to allow anonymous users
+        }
+
+        // Check if the current user is authenticated as per ESAPI Authenticator.
+        User currentUser = ESAPI.authenticator().getCurrentUser();
+
+        if ( ! currentUser.isAnonymous() ) {
+                // FIXME / OPENISSUE: @jeremiahjstacey - I think this should be set to 'debug'
+                // rather than 'info' level because I figured that the web / app
+                // server access logs should pick up the the authenticated user
+                // and log it along with whatever the HTTP request is,
+                // presumably including the authenticated user account name. However,
+                // I was unable to change 'src/test/resources/esapi-java-logging.properties'
+                // from 'INFO' level to 'DEBUG' level and then get log statement to show up in
+                // the JUnit test 'HTTPUtilitiesTest.java'. Any idea why? I
+                // thought that would work.
+            logger.info(Logger.SECURITY_SUCCESS, "Allowing authenticated user '" + currentUser.getAccountName() +
```

Comment:
The changes you're describing should have worked to record this at a debug level.  I can propose several possibilities that may have resulted in the change not applying correctly, but I'd want to step through a debugger to narrow it down a bit more.

I'm assuming this will not block the release, correct?

Reply:
Just to confirm, did you set both the overall level and the console logging levels?
```
handlers= java.util.logging.ConsoleHandler
.level= DEBUG
java.util.logging.ConsoleHandler.level = DEBUG
[.. other file stuff ..]
```

Reply:
Yes I did;I changed both. I even did a 'mvn clean compile test' too just in case needed to be copied somewhere under 'target/'.

Reply:
Ah, so that's it. Okay, I am just going to change it back to 'logger.debug' and then document this on the wiki page https://github.com/ESAPI/esapi-java-legacy/wiki/Configuration-Reference:-JavaLogFactory#diagnosing-common-problems